eclipse/src/saros/editor/internal/EditorAPI.java

```diff
@@ -291,6 +296,242 @@ public static OldTextSelection getOldSelection(IEditorPart editorPart) {
     return new OldTextSelection(jfaceSelection.getOffset(), jfaceSelection.getLength());
   }
 
+  /**
+   * Returns the line base selection values for the given editor part.
+   *
+   * <p>The given editor part must not be <code>null</code>.
+   *
+   * @param editorPart the editorPart for which to get the text selection
+   * @return the line base selection values for the given editor part or {@link
+   *     TextSelection#EMPTY_SELECTION} if the given editor part is not a text editor, does not have
+   *     a valid selection provider, or the correct line numbers or in-lin offsets could not be
+   *     calculated
+   * @throws NullPointerException if the given editor part is <code>null</code>
+   */
+  public static TextSelection getSelection(IEditorPart editorPart) {
+    Objects.requireNonNull(editorPart, "The given editor part must not be null");
+
+    if (!(editorPart instanceof ITextEditor)) {
+      return TextSelection.EMPTY_SELECTION;
+    }
+
+    ITextEditor textEditor = (ITextEditor) editorPart;
+    ISelectionProvider selectionProvider = textEditor.getSelectionProvider();
+
+    if (selectionProvider == null) {
+      return TextSelection.EMPTY_SELECTION;
+    }
+
+    IDocument document = textEditor.getDocumentProvider().getDocument(editorPart.getEditorInput());
+
+    ITextSelection textSelection = (ITextSelection) selectionProvider.getSelection();
```

Comment:
The [ITextSelection](https://help.eclipse.org/2019-12/index.jsp?topic=%2Forg.eclipse.platform.doc.isv%2Freference%2Fapi%2Forg%2Feclipse%2Fjface%2Ftext%2FITextSelection.html) provides methods as `getStartLine` is this another context?